src/main/java/org/broadinstitute/hellbender/tools/examples/ExampleTwoPassVariantWalker.java

```diff
@@ -0,0 +1,88 @@
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.ExampleProgramGroup;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.ReadsContext;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.engine.TwoPassVariantWalker;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.broadinstitute.hellbender.utils.variant.GATKVCFConstants.QUAL_BY_DEPTH_KEY;
+
+@CommandLineProgramProperties(
+        summary = "Example variant walker that makes two passes through a vcf",
+        oneLineSummary = "Example two variant walker",
+        programGroup = ExampleProgramGroup.class,
+        omitFromCommandLine = true
+)
+public class ExampleTwoPassVariantWalker extends TwoPassVariantWalker {
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc = "Output vcf", optional = true)
+    private File outputVcf = null;
+
+    final List<Double> qualByDepths = new ArrayList<>();
+
+    private VariantContextWriter vcfWriter;
+
+    public static String QD_DISTANCE_FROM_MEAN = "QD_DIST";
+
+    public static String COPY_OF_QD_KEY_NAME = "QD_COPY";
+
+    private double averageQualByDepth;
+
+    private double sampleVarianceOfQDs;
+
+    int counter = 0;
+
+    @Override
+    public void onTraversalStart() {
+        final VCFHeader inputHeader = getHeaderForVariants();
+        inputHeader.addMetaDataLine(new VCFInfoHeaderLine(QD_DISTANCE_FROM_MEAN, 1, VCFHeaderLineType.Float, "distance from the average QD value in the units of standard deviations"));
+        inputHeader.addMetaDataLine(new VCFInfoHeaderLine(COPY_OF_QD_KEY_NAME, 1, VCFHeaderLineType.Float, "copy of the QD INFO field"));
+        vcfWriter = createVCFWriter(outputVcf);
+        vcfWriter.writeHeader(inputHeader);
+    }
+
+    @Override
+    protected void firstPassApply(VariantContext variant, ReadsContext readsContext, ReferenceContext referenceContext, FeatureContext featureContext) {
+        qualByDepths.add(variant.getAttributeAsDouble(QUAL_BY_DEPTH_KEY, 0.0));
+    }
+
+    @Override
+    protected void secondPassApply(VariantContext variant, ReadsContext readsContext, ReferenceContext referenceContext, FeatureContext featureContext) {
+        VariantContextBuilder vcb = new VariantContextBuilder(variant);
+        final double qualByDepth = qualByDepths.get(counter++);
+        final double distanceFromMean = Math.abs((qualByDepth - averageQualByDepth)/ Math.sqrt(sampleVarianceOfQDs));
+        vcb.attribute(QD_DISTANCE_FROM_MEAN, distanceFromMean);
+        vcb.attribute(COPY_OF_QD_KEY_NAME, qualByDepth);
+
+        vcfWriter.add(vcb.make());
+    }
+
+    @Override
+    protected void afterFirstPass() {
```

Comment:
Example code would be more logical if you put `afterFirstPass()` physically between `firstPassApply()` and `secondPassApply()`

Reply:
done